roles/ceph-common/tasks/installs/install_on_suse.yml

```diff
@@ -5,7 +5,7 @@
 - name: Check for supported installation method on suse
   fail:
     msg: "Unsupported installation method origin:{{ ceph_origin }} repo:{{ ceph_repository }}'"
-  when: ceph_origin != 'distro' or (ceph_origin == 'repository' and ceph_repository != 'obs')
+  when: (ceph_origin != 'distro' and not (ceph_origin == 'repository' and ceph_repository == 'obs')) or (ceph_origin == 'repository' and ceph_repository != 'obs')
```

Comment:
I would either split it into two tasks:

```yaml
- name: Check ceph_origin definition on suse
  fail:
    msg: "unsupported ceph_origin ...."
  when: ceph_origin not in ['distro', 'repository']
```


```yaml
  - name: Check ceph_repository definition on suse
    fail:
      msg: "unsupported ceph_repository definition ...."
    when:
      - ceph_origin == 'repository'
      - ceph_repository != 'obs'
```

or if you absolutely want to do it in 1 task :

```yaml
  - name: Check for supported installation method on suse
    fail:
      msg: "Unsupported installation method origin:{{ ceph_origin }} repo:{{ ceph_repository }}'"
    when:
      - ceph_origin not in ['distro', 'repository']
        or (ceph_origin == 'repository' and ceph_repository != 'obs')
```

which is, IMO, a bit more readable

Reply:
I split this up into two tasks, you are right this is a lot easier to read and understand...